cs/src/core/Device/Devices.cs

```diff
@@ -26,9 +26,10 @@ public static class Devices
         /// <param name="logPath">Path to file that will store the log (empty for null device)</param>
         /// <param name="preallocateFile">Whether we try to preallocate the file on creation</param>
         /// <param name="deleteOnClose">Delete files on close</param>
-        /// <param name="capacity"></param>
+        /// <param name="capacity">The maximal number of bytes this storage device can accommondate, or CAPACITY_UNSPECIFIED if there is no such limit</param>
+        /// <param name="recoverDevice">Whether to recover device metadata from existing files</param>
         /// <returns>Device instance</returns>
-        public static IDevice CreateLogDevice(string logPath, bool preallocateFile = true, bool deleteOnClose = false, long capacity = CAPACITY_UNSPECIFIED)
+        public static IDevice CreateLogDevice(string logPath, bool preallocateFile = true, bool deleteOnClose = false, long capacity = CAPACITY_UNSPECIFIED, bool recoverDevice = false)
```

Comment:
That makes sense, but the goal was to unblock current users because the recovery path was causing slowdown and incorrectness in certain cases. The goal was to make "no recovery" as the default so current users default to that.